roles/common/templates/etc-hosts.j2

```diff
@@ -3,7 +3,7 @@
 
 # Mesos mappings
 {% for item in groups.all %}
-{{item}} {{mesos_hostname}} {{hostvars[item]['ansible_hostname']}}
+{{item}} {{hostvars[item]['ansible_hostname']}}.{{ansible_dns}} {{hostvars[item]['ansible_hostname']}}
```

Comment:
I am fine with merging this as is and then I can create a PR with the above fix later.